### ISSUE TITLE
tf.nn.static_rnn gives different sizes of output compared to input

### ISSUE THREAD

**Issue by PaulNicolasHunter** [assetgraph/assetgraph#820]
I'm trying to implement LSTM nets using tensorflow.

and i have a LSTMcell with the number of units = 4. 

`cell = tf.nn.rnn_cell.BasicLSTMCell(4, state_is_tuple=True)`

and my input is a **`2 X 44`**(data_hold) matrix, which i split into 4.

`inputs_series = tf.split(data_hold, 4, axis=1) `

so, each unit must be of size 2 X 11 right? 

when i check my input series it is a list of tensors of shape **`2 X 11`** as expected.

    [<tf.Tensor 'split:0' shape=(2, 11) dtype=float32>, <tf.Tensor 'split:1' shape=(2, 11) dtype=float32>, <tf.Tensor 'split:2' shape=(2, 11) dtype=float32>, <tf.Tensor 'split:3' shape=(2, 11) dtype=float32>]

but when i check the **`state_series`**, it is a list of tensors of shape **`2 X 4`**. 

    # initial_state is initial_state = tf.nn.rnn_cell.LSTMStateTuple(state_data_hold, hidden_data_hold)
    state_series, current_step = tf.nn.static_rnn(cell=cell, inputs=inputs_series, initial_state=initial_state)
   

     # state_series
    [<tf.Tensor 'rnn/rnn/basic_lstm_cell/mul_2:0' shape=(2, 4) dtype=float32>, <tf.Tensor 'rnn/rnn/basic_lstm_cell/mul_5:0' shape=(2, 4) dtype=float32>, <tf.Tensor 'rnn/rnn/basic_lstm_cell/mul_8:0' shape=(2, 4) dtype=float32>, <tf.Tensor 'rnn/rnn/basic_lstm_cell/mul_11:0' shape=(2, 4) dtype=float32>]

my question is that shouldn't the shape of the state series and the input series be same ? 

> As per the docs of static_rnn Returns: A pair (outputs, state) where:
> 
> outputs is a length T list of outputs (one for each input), or a
> nested tuple of such elements. state is the final state

and when i print current_step it returns a tuple of current and hidden unit, but which unit to be specific? (there are 4 right?)

any guesses?

**Comment by papandreou** [comment]
Wrong repo? :)

**Comment by PaulNicolasHunter** [comment]
sorry, i found the answer though
